src/runtime_src/core/pcie/linux/pcidev.cpp

```diff
@@ -455,21 +455,23 @@ open(const std::string& subdev, int flag) const
 }
 
 dev::
-dev(const drv& driver, const std::string& sysfs) : m_sysfs_name(sysfs), m_driver(driver)
+dev(std::shared_ptr<const drv> driver, const std::string& sysfs)
+  : m_sysfs_name(sysfs)
+  , m_driver(driver)
```

Comment:
```suggestion
dev(std::shared_ptr<const drv> driver, std::string sysfs)
  : m_sysfs_name(std::move(sysfs))
  , m_driver(std::move(driver))
```

Reply:
Can't make this change. Xbmgmt or xbutil will segfault. This is the constructor of pcidev object, caller may not expect that the arguments are std::move'd? Say, caller passes in a 'driver', to construct pcidev, then use the same 'driver' to do something else...

Reply:
Both arguments are passed by value.  `driver` you already passed by value and I changed `sysfs` to by value, so yes, you can and should move them into the data members.  By-value-and-move is btw the recommend way to initialize data member.

Reply:
You're right. It is the implementation of the constructor which is not ready for the by-value-and-move change. Now it's fixed.